File: AalsiTrader/Models/EquityPoint.swift
import Foundation

struct EquityPoint: Codable, Identifiable {
    var id: String { "\(timestamp)" }
    let timestamp: Double
    let capital: Double
    let pnl: Double
    let drawdown: Double
    let openPositions: Int?
    var formattedTime: String?
    var formattedDate: String?

    var date: Date {
        Date.fromEpoch(timestamp)
    }
}

File: AalsiTraderAndroid/app/src/main/java/com/aalsitrader/android/model/EquityPoint.kt
package com.aalsitrader.android.model

import kotlinx.serialization.Serializable

@Serializable
data class EquityPoint(
    val timestamp: Double = 0.0,
    val capital: Double = 0.0,
    val pnl: Double = 0.0,
    val drawdown: Double = 0.0,
    val openPositions: Int? = null,
    val formattedTime: String? = null,
    val formattedDate: String? = null,
)
